heat/core/logical.py

```diff
@@ -307,3 +314,55 @@ def logical_xor(t1, t2):
     tensor([ False, False,  True])
     """
     return operations.__binary_op(torch.logical_xor, t1, t2)
+
+
+def _sanitize_close_input(x, y):
+    if np.isscalar(x):
+        try:
+            if isinstance(y, dndarray.DNDarray):
+                x = factories.array(float(x), device=y.device)
+            else:
+                x = factories.array(float(x))
+        except (ValueError, TypeError):
+            raise TypeError("Data type not supported, input was {}".format(type(x)))
+
+    elif not isinstance(x, dndarray.DNDarray):
+        raise TypeError(
+            "Only tensors and numeric scalars are supported, but input was {}".format(type(x))
+        )
+
+    if np.isscalar(y):
+        try:
+            if isinstance(x, dndarray.DNDarray):
+                y = factories.array(float(y), device=x.device)
+            else:
+                y = factories.array(float(y))
+        except (ValueError, TypeError):
+            raise TypeError("Data type not supported, input was {}".format(type(y)))
+
+    elif not isinstance(y, dndarray.DNDarray):
+        raise TypeError(
+            "Only tensors and numeric scalars are supported, but input was {}".format(type(y))
+        )
+
+    # Do redistribution out-of-place
+    # If only one of the tensors is distributed, unsplit/gather it
+    if (x.split is not None) and (y.split is None):
```

Comment:
Are these parenthesis really pep-8 conform? Do you have the black commit hook installed?

Reply:
haha, I missed this. I do have the black commit hook installed as far as I can tell, and it hasn't complained, but I've removed the parentheses.